From bad776e1804047728fe3b8e8b48bab9bc21158b7 Mon Sep 17 00:00:00 2001
From: Amir Vadai <avadai@redhat.com>
Date: Thu, 11 Sep 2014 11:59:08 -0400
Subject: [infiniband] core: Fix unused variable warning

Message-id: <f7a603f1c4c938973fc5937819dbe946d6b74e2f.1410425016.git.avadai@redhat.com>
Patchwork-id: 93584
O-Subject: [Patch RHEL7.1 V1 012/197] IB/core: Fix unused variable warning
Bugzilla: 1107617 1107618 1107619 1107620
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>

From: Or Gerlitz <ogerlitz@mellanox.com>

Fix the below "make W=1" build warning:

    drivers/infiniband/core/iwcm.c: In function 'destroy_cm_id':
    drivers/infiniband/core/iwcm.c:330: warning: variable 'ret' set but not used

Signed-off-by: Or Gerlitz <ogerlitz@mellanox.com>
Signed-off-by: Roland Dreier <roland@purestorage.com>
(cherry picked from commit 05633102d85b50f35325dfbedafcedd6c5b3264c)
Signed-off-by: Amir Vadai <avadai@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/infiniband/core/iwcm.c b/drivers/infiniband/core/iwcm.c
index 0717940..3d2e489 100644
--- a/drivers/infiniband/core/iwcm.c
+++ b/drivers/infiniband/core/iwcm.c
@@ -334,7 +334,6 @@ static void destroy_cm_id(struct iw_cm_id *cm_id)
 {
  struct iwcm_id_private *cm_id_priv;
  unsigned long flags;
- int ret;
 
  cm_id_priv = container_of(cm_id, struct iwcm_id_private, id);
  /*
@@ -350,7 +349,7 @@ static void destroy_cm_id(struct iw_cm_id *cm_id)
   cm_id_priv->state = IW_CM_STATE_DESTROYING;
   spin_unlock_irqrestore(&cm_id_priv->lock, flags);
   /* destroy the listening endpoint */
-  ret = cm_id->device->iwcm->destroy_listen(cm_id);
+  cm_id->device->iwcm->destroy_listen(cm_id);
   spin_lock_irqsave(&cm_id_priv->lock, flags);
   break;
  case IW_CM_STATE_ESTABLISHED:
-- 
1.7.1